docs/gen_ref_pages.py

```diff
@@ -1,111 +1,58 @@
 """
 Generate the code reference pages.
 
-Source: https://mkdocstrings.github.io/recipes/
+This script traverses the markdown files under docs/src/reference and for each:
+- If the file is not empty it will appear in the docs as-is (after mkdocstrings
+    has filled in any identifiers inside it).
+- If the file is empty, it is interpreted as a placeholder for the top-level
+    symbol with its same name and its reference page is generated in its same
+    path. E.g., if an empty docs/src/reference/temporian/io/to_csv.md file
+    exists, the reference page for `temporian.to_csv` will be generated under
+    reference/temporian/io/to_csv/ in the docs.
+
+Related: https://mkdocstrings.github.io/recipes/
 """
 
 from pathlib import Path
-from typing import Set, Tuple
 
 import mkdocs_gen_files
 
 nav = mkdocs_gen_files.Nav()
 
-SRC_PATH = Path("temporian")
+REFERENCE = Path("docs/src/reference")
 
-# Stores symbol and path of each public API member
-members: Set[Tuple[str, Path]] = set()
+# Sort files by depth and name for alphabetical order and subdirs at the bottom
+paths = sorted(
+    REFERENCE.rglob("*.md"), key=lambda p: str(len(p.parts)) + str(p)
+)
 
-non_parsable_imports = []
+non_empty_files = []
 
-# We need to be able to parse other files to allow wildcard imports
-# Storing pair of (prefix, path) to parse in a stack
-files_to_parse = [(None, SRC_PATH / "__init__.py")]
+for path in paths:
```

Comment:
Would it be possible to check that all empty files point to a class / function ?

Reply:
mkdocstrings will fail if any of them doesn't 👍🏼